sound/soc/sof/sof-priv.h

```diff
@@ -582,7 +587,8 @@ void sof_print_oops_and_stack(struct snd_sof_dev *sdev, const char *level,
 			      u32 panic_code, u32 tracep_code, void *oops,
 			      struct sof_ipc_panic_info *panic_info,
 			      void *stack, size_t stack_words);
-int snd_sof_init_trace_ipc(struct snd_sof_dev *sdev);
+int snd_sof_trace_suspend(struct snd_sof_dev *sdev, int pm_state);
+int snd_sof_trace_resume(struct snd_sof_dev *sdev);
```

Comment:
You introduce more than new dtrace states.

Reply:
Hmm, true. Not so easy to write the commit message for this one. Maybe something like:

```
ASoC: SOF: extend the interface to stop DMA trace

Change the interface to stop the DMA trace for suspend. Replace the
snd_sof_init_trace_ipc() and snd_sof_release_trace() calls with more 
explicit interface for PM (the sole user for this interface). 

The new snd_sof_trace_suspend() call takes the target PM state as argument,
allowing the trace implementation to decide how to handle the transition.
Use this information to release DMA resources only if DSP is suspended and
will not remain in D0.
```